triedb/database/database.go

```diff
@@ -37,3 +38,31 @@ type NodeDatabase interface {
 	// An error will be returned if the specified state is not available.
 	NodeReader(stateRoot common.Hash) (NodeReader, error)
 }
+
+// StateReader wraps the Account and Storage method of a backing state reader.
+type StateReader interface {
+	// Account directly retrieves the account associated with a particular hash in
+	// the slim data format. An error will be returned if the read operation exits
+	// abnormally. Specifically, if the layer is already stale.
+	//
+	// Note:
+	// - the returned account object is safe to modify
+	// - no error will be returned if the requested account is not found in database
+	Account(hash common.Hash) (*types.SlimAccount, error)
+
+	// Storage directly retrieves the storage data associated with a particular hash,
+	// within a particular account. An error will be returned if the read operation
+	// exits abnormally.
+	//
+	// Note:
+	// - the returned storage data is not a copy, please don't modify it
+	// - no error will be returned if the requested slot is not found in database
+	Storage(accountHash, storageHash common.Hash) ([]byte, error)
+}
+
+// StateDatabase warps the methods of a backing state store.
```

Comment:
it `wraps` rather than warps, right :) 
```suggestion
// StateDatabase wraps the methods of a backing state store.
```